mantidproject/mantid
# Fix plot script generation for Bin plots

**Description of work.**
This PR fixes the plot script generation for figures constructed from a particular bin.



**To test:**
- Load some data
- Right-click and click show data
- Click on a column of the data
- Right click and do plot bin (with and without errors)
- Generate the plot script
- **You'll have to close the old figure as we recreate the script with the same `num` argument, meaning the figure from the script will overwrite the existing one. This is how the script generation has always functioned, so wondering whether we should change it..**
- Run the script
- Open the figure you created directly from the figure
- Compare the two, they should match
- Check existing functionality works



Fixes #30139 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
